### PR TITLE
use DNS discovery as default bootstrap discovery

### PR DESCRIPTION
**Pull request by danisharora099**
## Problem

ref: https://github.com/waku-org/js-waku/issues/517

## Solution

moves away from predefined list of nodes, and uses DNS discovery as bootstrap discovery

### CODE REVIEW

**Review by fryorcraken — CHANGES_REQUESTED** [review]
Are we ready for that?

Todo list because we can consider this change:

1. Test DNS Discovery with an example
2. Ask a number of people to use said example to do some dog fooding

**Comment by danisharora099** [comment]
> Are we ready for that?
> 
> Todo list because we can consider this change:
> 
>     1. Test DNS Discovery with an example

I've added 3 tests for the same:
1. libp2p-compliance-test
2. passing `dnsDiscovery` in `peerDiscovery` with light node
3. retrieving multiaddrs for `test.waku.nodes.status.im`

You can find the tests in `packages/tests/tests/dns-peer-discovery.spec.ts`


> 
>     2. Ask a number of people to use said example to do some dog fooding
Until we release these changes to npm, we can't consume them in our example app unless we build manually.

**Comment by danisharora099** [comment]
I've refactored the tests to be more explicit and added usage of `components` while initializing the discovery while tagging peers.

**Comment by fryorcraken** [comment]
Let me rephrase: we need to **dogfood** dns discovery before we make it the default discovery strategy.

To dogfood it, I suggest to make the `web-chat` example use it and ask colleagues to use the examples and let us know any feedback.

**Comment by danisharora099** [comment]
> 

Gotcha. I misinterpreted. 

Thanks for explaining. I should be able to integrate it with `web-chat` once we can release latest `master` to NPM.
Holding off on this PR until then.

File: packages/dns-discovery/src/index.ts [hunk 1]
```diff
@@ -15,6 +15,39 @@ export { NodeCapabilityCount };
 
 const log = debug("waku:peer-discovery-dns");
 
+const publicKey = "AOGECG2SPND25EEFMAJ5WF3KSGJNSGV356DSTL2YVLLZWIV6SAYBM";
+const fqdn = "test.waku.nodes.status.im";
+const enrTree = `enrtree://${publicKey}@${fqdn}`;
```
Comment (fryorcraken) [inline packages/dns-discovery/src/index.ts:20]: Var split is not necessary, just save it in one string and it'll help with cspell

File: .cspell.json [hunk 2]
```diff
@@ -111,7 +111,8 @@
     "webfonts",
     "websockets",
     "wifi",
-    "xsalsa20"
+    "xsalsa20",
+    "AOGECG2SPND25EEFMAJ5WF3KSGJNSGV356DSTL2YVLLZWIV6SAYBM"
```
Comment (fryorcraken) [inline .cspell.json:115]: use `patterns` to exclude enrtrees

File: packages/create/package.json [hunk 3]
```diff
@@ -68,6 +68,7 @@
     "@libp2p/mplex": "^7.0.0",
     "@libp2p/websockets": "^5.0.0",
     "@waku/core": "*",
+    "@waku/dns-discovery": "*",
     "@waku/peer-exchange": "*",
```
Comment (fryorcraken) [inline packages/create/package.json:72]: @libp2p/bootstrap should be removed

File: packages/create/src/index.ts [hunk 4]
```diff
@@ -55,6 +64,11 @@ export interface CreateOptions {
    * Use recommended bootstrap method to discovery and connect to new nodes.
    */
   defaultBootstrap?: boolean;
+
+  /**
+   * Node requirements to setup discovery with
+   */
+  nodeRequirements?: Partial<NodeCapabilityCount>;
```
Comment (fryorcraken) [inline packages/create/src/index.ts:71]: Not sure this is necessary, if a dev wants to fine tune their discovery experience, then they should not use default bootstrap.

File: packages/dns-discovery/src/index.ts [hunk 5]
```diff
@@ -58,7 +89,15 @@ export class PeerDiscoveryDns
...
-      peerInfos.forEach((peerInfo) => {
+      peerInfos.forEach(async (peerInfo) => {
+        await this._components.peerStore.tagPeer(
+          peerInfo.id,
+          DEFAULT_BOOTSTRAP_TAG_NAME,
+          {
+            value: this._options.tagValue ?? DEFAULT_BOOTSTRAP_TAG_VALUE,
+            ttl: this._options.tagTTL ?? DEFAULT_BOOTSTRAP_TAG_TTL,
+          }
+        );
```
Comment (fryorcraken) [inline packages/dns-discovery/src/index.ts:100]: What are these changes? If more modification to dns discovery is necessary then lets' do that and dog food that first.

File: packages/tests/tests/dns-peer-discovery.spec.ts [hunk 6]
```diff
@@ -1,20 +1,38 @@
...
   this.timeout(10000);
   tests({
     async setup() {
-      return wakuDnsDiscovery(enrTree, {
-        filter: 1,
-      })();
+      // create libp2p mock peerStore
+      const components = new Components({
+        peerStore: new PersistentPeerStore({
+          peerId: await createEd25519PeerId(),
```
Comment (fryorcraken) [inline packages/tests/tests/dns-peer-discovery.spec.ts:25]: Waku only use secp256k1 peer id

File: packages/create/src/index.ts [hunk 7]
```diff
@@ -14,14 +13,24 @@ import {
...
 import type { FullNode, IRelay, LightNode, RelayNode } from "@waku/interfaces";
 import { wakuPeerExchange } from "@waku/peer-exchange";
 import type { Libp2p } from "libp2p";
 import { createLibp2p, Libp2pOptions } from "libp2p";
 
 import type { Libp2pComponents } from "./libp2p_components.js";
 
+const DEFAULT_NODE_REQUIREMENTS = {
+  lightPush: 2,
+  filter: 2,
```
Comment (fryorcraken) [inline packages/create/src/index.ts:30]: Does this assume peer exchange works? if not, then should be 1 everywhere.

File: packages/dns-discovery/src/index.ts [hunk 8]
```diff
@@ -15,6 +15,37 @@ export { NodeCapabilityCount };
 
 const log = debug("waku:peer-discovery-dns");
 
+const enrTree = `enrtree://AOGECG2SPND25EEFMAJ5WF3KSGJNSGV356DSTL2YVLLZWIV6SAYBM@test.waku.nodes.status.im`;
```
Comment (fryorcraken) [inline packages/dns-discovery/src/index.ts:18]: Sorry missed that earlier. Two changes:
1. Have both values for wakuv2.prod and wakuv2.test fleet available
2. Use wakuv2.prod fleet value by default (not test)

**Review by fryorcraken — APPROVED** [review]
